protodf/__init__.py

```diff
@@ -45,19 +44,24 @@ def message_to_row(descriptor, message):
     for field_tuple in message.ListFields():
         field_map[field_tuple[0].name] = field_tuple[1]
 
-    values = []
+    values = {}
+    for field_descriptor in sorted(descriptor.fields, key=lambda x: x.name):
+        values[field_descriptor.name] = __get_field_value(field_descriptor, field_map)
+
+    return Row(**values)
 
-    for field_descriptor in descriptor.fields:
-        if field_descriptor.name in field_map:
-            if field_descriptor.label == field_descriptor.LABEL_REPEATED:
-                values.append([__to_row_data(field_descriptor, data) for data in
-                               field_map[field_descriptor.name]])
-            else:
-                values.append(__to_row_data(field_descriptor, field_map[field_descriptor.name]))
-        else:
-            values.append(None)
 
-    return Row(*values)
+def __get_field_value(field_descriptor, field_map):
+    if field_descriptor.name not in field_map:
+        return None
+
+    if isinstance(field_map[field_descriptor.name], Mapping):
+        return [Row(**{'key': k, 'value': v}) for k, v in field_map[field_descriptor.name].items()]
```

Comment:
A map value can be any type, so shouldn't there be a recursive call like `__to_row_data(v)`?

https://developers.google.com/protocol-buffers/docs/proto3#maps

Reply:
Great catch! :)
pushed a commit to fix that